### PR TITLE
Remove methods that were moved to client repo

### PR DESCRIPTION
**Pull request by Marenz**
The PR looks large, but most commits are just small changes. Tried to keep it well separated.

### CODE REVIEW

File: src/frequenz/dispatch/_dispatcher.py [hunk 1]
```diff
@@ -210,6 +210,17 @@ async def start(self) -> None:
         """Start the actor."""
         self._actor.start()
 
+    async def resend_current_running_states(self, dispatch_type: str) -> None:
```
Comment (Marenz) [inline src/frequenz/dispatch/_dispatcher.py:213]: Maybe this? @llucax @ela-kotulska-frequenz

**Review by llucax — DISMISSED** [review]
Bye bye `RunningState.DIFFERENT_TYPE`, you will not be missed :rofl: :heart:

**Comment by Marenz** [comment]
> bye bye RunningState.DIFFERENT_TYPE, you will not be missed 

I missed it already, tbh. Now you have to make double-sure you check for the .type before looking at `running`. Already missed one occasion for that before that I luckily found later.

Still not convinced this is a better interface that leads to less errors.

**Comment by Marenz** [comment]
Maybe the receiver should come pre-filtered for a set of types you specify :thinking:

**Pull request merged**